### PR TITLE
FIX: abortMpu returns 204 if no such uploadId

### LINKED ISSUE #568: Abort MPU returns 404 for all regions when Upload Id doesn't exist
Tests reveal that AWS S3 returns 404 for us-east-1 and 204 for all other regions. We have a customer complaining about this incorrect behavior. We could leverage `usEastBehavior` to toggle the legacy behavior.
The fix should be against `6.2.5`.

### PR DESCRIPTION
**Pull request by electrachong**
FIX #568

### CODE REVIEW

File: tests/unit/api/multipartDelete.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                query: `uploadId=${testUploadId}`,
+            };
+            next(null, deleteMpuRequest);
+        },
+        (deleteMpuRequest, next) =>
+            multipartDelete(authInfo, deleteMpuRequest, log, next),
+    ], callback);
+}
+
+describe.only('Multipart Delete API', () => {
```
Comment (LaurenSpiegel) [inline tests/unit/api/multipartDelete.js]: nix the only

File: tests/unit/api/multipartDelete.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    beforeEach(() => {
+        cleanup();
+    });
+    afterEach(() => {
+        cleanup();
+    });
+
+    it('should return 204 if uploadId does not exist', done => {
```
Comment (LaurenSpiegel) [inline tests/unit/api/multipartDelete.js]: should return 204 if uploadId does not exist on multipart abort call

File: lib/api/multipartDelete.js [hunk 3]
```diff
@@ -57,7 +58,16 @@ function multipartDelete(authInfo, request, log, callback) {
                 });
         },
         function checkMPUval(next) {
-            services.metadataValidateMultipart(metadataValParams, next);
+            services.metadataValidateMultipart(metadataValParams,
+                (err, mpuBucket, mpuOverviewArray) => {
+                    if (err === errors.NoSuchUpload) {
+                        log.trace('did not find valid mpu with uploadId' +
+                        `${uploadId}`, { method: 'multipartDelete' });
+                        // return 404 if usEastBehavior is enabled in config
```
Comment (LaurenSpiegel) [inline lib/api/multipartDelete.js]: TODO in rel/6.3 and later: return 404 if usEastBehavior is enabled in config

File: tests/unit/api/multipartDelete.js (hunk 2 above)
Comment (electrachong) [inline tests/unit/api/multipartDelete.js]: ok

File: lib/api/multipartDelete.js (hunk 3 above)
Comment (electrachong) [inline lib/api/multipartDelete.js]: ok

File: tests/unit/api/multipartDelete.js (hunk 1 above)
Comment (electrachong) [inline tests/unit/api/multipartDelete.js]: 👍

**Comment by electrachong** [comment]
@ironman-machine

**Comment by electrachong** [comment]
end-to-end: http://ci.ironmann.io/gh/scality/Integration/9286

**Pull request merged**